src/gridcoin/claim.h

```diff
@@ -187,6 +187,19 @@ class Claim : public IContractPayload
     //!
     SuperblockPtr m_superblock;
 
+    //!
+    //! \brief A map of mrc transactions keyed by CPID. There can be only one MRC per CPID
+    //! in the claim.
+    //!
+    std::map<Cpid, uint256> m_mrc_tx_map;
```

Comment:
Consider `unordered_map` if order doesn't matter.

Reply:
Probably better, although I have to refork my testnet to test it, since it is a structural change in the claim. There can be at most 5 elements in this map, so I am not sure it really matters for a map that small.

Reply:
Do we really care that much?

Reply:
The order does not matter. You can't store output offsets (index numbers) in the claim because the claim is created BEFORE the splitting/sidestake of the main research reward, so the index of the MRC outputs can shift. We need to maintain the order of the offsets the way they are to minimize changes to other parts of the code... empty... main stake return (and splits) ... sidestakes ... (foundation mrc fee sidestake) ... mrc payments.

Reply:
This requires us to walk the map in the ValidateMRCRewards function when we verify the correlation of the outputs on the coinstake to the MRCs bound in the claim, but again the number of elements is extremely small on the map.